src/clj/game/cards/events.clj

```diff
@@ -40,38 +40,37 @@
-    :msg "trash all installed Corp cards and turn all installed Runner cards facedown"
-    :effect (req (let [ai (all-installed state :corp)
-                       onhost (filter #(= '(:onhost) (:zone %)) ai)
-                       allcorp (->> ai
-                                    (remove #(= '(:onhost) (:zone %)))
-                                    (sort-by #(vec (:zone %)))
-                                    (reverse))]
-                   ; Trash hosted cards first so they don't get trashed twice
-                   (doseq [c onhost]
-                     (trash state side c))
-                   (doseq [c allcorp]
-                     (trash state side (get-card state c))))
-
-                 ;; do hosted cards first so they don't get trashed twice
-                 (let [installedcards (all-active-installed state :runner)
-                       ishosted (fn [c] (or (= ["onhost"] (get c :zone)) (= '(:onhost) (get c :zone))))
-                       hostedcards (filter ishosted installedcards)
-                       nonhostedcards (remove ishosted installedcards)]
-                   (doseq [oc hostedcards :let [c (get-card state oc)]]
-                     (move state side c [:rig :facedown])
-                     (if (:memoryunits c)
-                       (gain state :runner :memory (:memoryunits c))))
-                   (doseq [oc nonhostedcards :let [c (get-card state oc)]]
-                     (move state side c [:rig :facedown])
-                     (if (:memoryunits c)
-                       (gain state :runner :memory (:memoryunits c))))))}
+   (let [corp-trash {:delayed-completion true
+                     :effect (req (let [ai (all-installed state :corp)
+                                        onhost (filter #(= '(:onhost) (:zone %)) ai)
+                                        unhosted (->> ai
+                                                     (remove #(= '(:onhost) (:zone %)))
+                                                     (sort-by #(vec (:zone %)))
+                                                     (reverse))
+                                        allcorp (concat onhost unhosted)] 
+                                    (trash-cards state :runner eid allcorp)))}
+         runner-facedown {:effect (req (let [installedcards (all-active-installed state :runner)
+                                             ishosted (fn [c] (or (= ["onhost"] (get c :zone)) (= '(:onhost) (get c :zone))))
+                                             hostedcards (filter ishosted installedcards)
+                                             nonhostedcards (remove ishosted installedcards)]
+                                         (doseq [oc hostedcards :let [c (get-card state oc)
+                                                                      c (deactivate state side c true)
+                                                                      c (assoc-in c [:facedown] true)]]
```

Comment:
Why is `move` unsuitable for these cards?

If `move` doesn't work, then It's probably best if these effects go into a function that can be reused, in case we get future effects that turn installed cards facedown.

Reply:
These cards are hosted (and should remain so), move does not handle moving to hosted position.

Can make a `flip-facedown` function, that checks if a card is hosted, and either does this or moves it in the normal manner, np.